File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum = 0
        max_num = nums[0]
        for item in nums:
            sum += item
            max_num = max(sum, max_num)
            if sum < 0:
                sum = 0

        return max_num
    

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let sum = 0;
    max_num = nums[0];
    for (const item of nums) {
        sum += item
        max_num = Math.max(sum, max_num)
        if (sum < 0){
            sum = 0;
        };
    };
    return max_num;
};